### ISSUE TITLE
TypeError: string indices must be integers

### ISSUE THREAD

**Issue by Outcats** [evilhero/mylar#1879]
Full error message:

> Error !
> TypeError: string indices must be integers
> <tbody> %if searchresults: %for result in searchresults: <% if result['haveit'] == 'No': haveit = 'No' else: haveit = result['haveit']['comicid'] grade = 'Z' 
> S:/Websites/Mylar/data/interfaces/default/searchresults.html, line 50:
> <%
> S:/Websites/Mylar/data/interfaces/default/base.html, line 87:
> ${next.body()}
> S:\Websites\Mylar\lib\mako\runtime.py, line 899:
> callable_(context, *args, **kwargs)
> S:\Websites\Mylar\lib\mako\runtime.py, line 873:
> _exec_template(inherit, lclcontext, args=args, kwargs=kwargs)
> S:\Websites\Mylar\lib\mako\runtime.py, line 838:
> **_kwargs_for_callable(callable_, data))
> S:\Websites\Mylar\lib\mako\template.py, line 462:
> return runtime._render(self, self.callable_, args, data)
> S:\Websites\Mylar\mylar\webserve.py, line 54:
> return template.render(http_root=mylar.CONFIG.HTTP_ROOT, **kwargs)

I am getting this in the Story Arc section when searching for a new story arc to add. This particular one is "Civil War", I was playing around and found if I put in something to shorten the number of results such as "Civil War 2" it brought them back so it might be a number of results issue? (I am specifically looking for the original Civil War arc so can't really shorten it any more than that.)

I am not seeing anything in the Python read out when this error comes up on the site.

**Comment by evilhero** [comment]
The result is in the wrong format for it to understand. The format was changed a few months ago, and I guess this one straggled by somehow - so I've fixed it now so it won't throw that error anymore. As well, I also fixed the Story Arc Year not being populated on the search results page. Commit up shortly into dev for all of this...

**Comment by Outcats** [comment]
Hey, I've cloned the development branch and copied my config.ini and db files over to it, ran the same search and got the same error. I've confirmed the new lines in those two files you linked in your commit were in the files I downloaded.

Here is the message again, i'm 99% sure it is the same error.
![image](https://user-images.githubusercontent.com/11628565/37094819-12fc96e2-220c-11e8-94ed-567d340b020f.png)

**Comment by evilhero** [comment]
Do you have the Civil War arc currently in your story arc list ? It does take about a minute for the results to return to the screen as it has to calculate the amount of issues in each arc - but I just tried it on the dev branch and the results are coming back fine and the search results screen displays without an error on this end.

If you enable debug logging in Mylar, is there nothing in there when it's loading up the search results ?

**Comment by Outcats** [comment]
I do have it in my list, basically I accidentally removed a few issues from my arc when trying to manually get the system to search for them as I am unable to just search all missing or mark them missing all wanted at once. I was hoping I could re-add the arc without removing my existing list to refresh the list. Is there a way to add a reset to the arc list by chance?

You saying if I remove my existing civil war arc and search it should be fine?

**Comment by evilhero** [comment]
I honestly don't know at this point - the fact that it's working for me and not for you, would point to something in your db might be strangling it somehow, or it might not. I had the arc in mine at one point, I searched and the error occurred. I patched the problem via the commit. Then I removed the arc, and searched and it worked. Thereafter, I added the arc again, searched and it didn't have any errors.

Have you tried to just refresh the list via the main story arc page? It will however retain your deleted items, so if you want to get those issues back - you'd need to delete the arc and re-add it so it knows that it's supposed to be included. That or manually add the ones you deleted back in via the Add option in the lower right corner of the arc page, but that might get tedious.
